src/user-theme.ts

```diff
@@ -1,71 +1,59 @@
-import { parse } from "jsonc-parser";
-import { getVscode } from "./vscode-utils.js";
-import type { SpecialTheme, ThemeRegistrationAny } from "shiki/types";
+import type { SpecialTheme, ThemeRegistration } from "shiki";
+import { ThemeRegistry } from "./theme-registry.js";
+import { getVscode, ExtensionFileReader } from "./vscode-utils.js";
+import { logger } from "./logger.js";
+import { buildThemeRegistration } from "./shiki-bridge-theme.js";
+
+export type UserThemeResult = [id: string, theme: ThemeRegistration] | [id: string, theme: SpecialTheme];
+
+let cache: ThemeRegistry | null = null;
+function getThemeRegistry(vscode: typeof import('vscode')): ThemeRegistry {
+  if (!cache) {
+    cache = ThemeRegistry.build(vscode.extensions.all);
```

Comment:
I think we shouldn't load all the extensions to the registry rather than load the user language and it's dependencies

Reply:
Make sense too